pkg/cluster/delete.go

```diff
@@ -63,3 +81,47 @@ func Delete(
 	}
 	return nil
 }
+
+func DeleteOverridesDir(app *v1alpha1.Application, kubeClient *kubernetes.Clientset, argocdNs string, clusterName string) error {
+	log := logpkg.GetLogger()
+	repoUrl := app.Annotations[baseClusterRepoUrlAnnotation]
+	repoRevision := app.Annotations[baseClusterRepoRevisionAnnotation]
+	repoPath := app.Annotations[baseClusterRepoPathAnnotation] + "/" + clusterName
+	creds, err := argocd.GetRepoCredsFromArgoCd(kubeClient, argocdNs, repoUrl)
+	repo, tmpDir, auth, err := argocd.CloneRepo(creds, repoUrl, repoRevision)
+	if err != nil {
+		return fmt.Errorf("failed to clone repo: %s", err)
+	}
+	wt, err := repo.Worktree()
+	fileInfo, err := wt.Filesystem.Lstat(repoPath)
+	if err == nil {
+		if !fileInfo.IsDir() {
+			return fmt.Errorf("unexpected file type for %s", repoPath)
+		}
+		_, err := wt.Remove(repoPath)
+		if err != nil {
+			return fmt.Errorf("failed to recursively delete cluster directory: %s", err)
+		}
+	}
```

Comment:
What if err != nil?

Reply:
Handled all these errors in the recent change